hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicator.java

```diff
@@ -302,6 +350,12 @@ private List<Future<Boolean>> replicateTable(
     int numCellsInBatch = 0;
     for (Map.Entry<ByteRange, List<Cell>> rowCells : cellsToReplicateByRow.entrySet()) {
 
+      // If two-way replication is enabled, tag row cells with origin info so Bigtable CDC knows to
+      // skip this batch and not replicate batch back to HBase.
+      if (isTwoWayReplication) {
+        appendOriginInfoToRowCells(rowCells);
```

Comment:
how about addSourceTag or addOriginTag? Also use the term "source/origin tagging" everywhere in the bidirectional replication docs.

@igorbernstein2 what do you think is better "source tagging" or "origin tagging"?

Reply:
Using source tagging for now